datacite-draft-doi-handler/src/test/java/no/unit/nva/doi/DraftDoiHandlerTest.java

```diff
@@ -0,0 +1,110 @@
+import java.io.ByteArrayOutputStream;
+import java.io.InputStream;
+import java.nio.file.Path;
+import no.unit.nva.doi.model.DoiUpdateDto;
+import no.unit.nva.events.models.AwsEventBridgeDetail;
+import no.unit.nva.events.models.AwsEventBridgeEvent;
+import no.unit.nva.publication.doi.dto.Publication;
+import no.unit.nva.publication.doi.dto.PublicationDtoTestDataGenerator;
+import no.unit.nva.publication.doi.dto.PublicationHolder;
+import nva.commons.utils.IoUtils;
+import nva.commons.utils.JsonUtils;
+import org.hamcrest.MatcherAssert;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class DraftDoiHandlerTest {
+
+    public static final String DOI_PUBLICATION = "doi.publication";
+    public static final String ERROR_MESSAGE = "error";
+
+    private DraftDoiHandler handler;
+    private ByteArrayOutputStream outputStream;
+    private Context context;
+    private AwsEventBridgeEvent<AwsEventBridgeDetail<PublicationHolder>> event;
+
+    @BeforeEach
+    public void setUp() {
+        handler = new DraftDoiHandler();
+        outputStream = new ByteArrayOutputStream();
+        context = mock(Context.class);
+        event = mock(AwsEventBridgeEvent.class);
+    }
+
+
+    @Test
+    public void handleRequestReturnsDoiUpdateDtoWithPublicationUriWhenInputIsValid() {
+        InputStream inputStream = IoUtils.inputStreamFromResources(
+            Path.of("doi_publication_event_valid.json"));
+        handler.handleRequest(inputStream,outputStream,context);
+        DoiUpdateDto response = parseResponse();
+        assertThat(response.getPublicationId(), is(not(nullValue())));
+    }
+
+    private DoiUpdateDto parseResponse() {
+        return attempt(() -> JsonUtils.objectMapper.readValue(outputStream.toString(), DoiUpdateDto.class))
+            .orElseThrow();
+    }
+
+    @Test
+    public void handleRequestThrowsExceptionOnMissingEventItem() {
+        InputStream inputStream = IoUtils.inputStreamFromResources(
+            Path.of("doi_publication_event_empty_item.json"));
+        IllegalArgumentException exception = assertThrows(IllegalArgumentException.class,
+            () -> handler.handleRequest(inputStream, outputStream, context));
+
+        MatcherAssert.assertThat(exception.getMessage(), is(DraftDoiHandler.PUBLICATION_IS_MISSING_ERROR));
+    }
+
+    @Test
+    public void handleRequestThrowsExceptionOnMissingCustomerId() {
+        InputStream inputStream = IoUtils.inputStreamFromResources(
+            Path.of("doi_publication_event_empty_institution_owner.json"));
+        IllegalArgumentException exception = assertThrows(IllegalArgumentException.class,
+            () -> handler.handleRequest(inputStream, outputStream, context));
+
+        MatcherAssert.assertThat(exception.getMessage(), is(DraftDoiHandler.CUSTOMER_ID_IS_MISSING_ERROR));
+    }
+
+    @Test
+    public void draftDoiHandlerReturnsDoiUpdateDtoOnValidPublicationDto() {
```

Comment:
This test we can delete, too close to implementation details right?

Reply:
Or rename to `handleRequestReturnsDoiUpdateDtoWhenValidPublicationHolder` ? 

We probably want a createInputStream and assemble a event where we use the Publication holder from the test data generator. 

Anyhow, I do agree that the current PublicationDtoTestGenerator is not ideally how we want it probably.